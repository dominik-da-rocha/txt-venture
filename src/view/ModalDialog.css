.ModalDialog {
  position: fixed;
  display: flex;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #0000005e;
  z-index: 1000;
}

.ModalDialog button {
  user-select: none;
}

.ModalDialog .Dialog {
  background: white;
  margin: auto;
  min-width: 10rem;
  min-height: 5rem;
  box-shadow: 1px 1px 5px black;
}

.ModalDialog .Dialog .Header {
  display: flex;
  justify-content: space-between;
  background: #286edd;
  color: white;
  padding: 0.25rem;
}

.ModalDialog .Dialog .Header .X {
  user-select: none;
  padding: 0.1rem;
  width: 1.5rem;
  height: 1rem;
  text-align: center;
}

.ModalDialog .Dialog .Header .X:hover {
  background: #c4320d;
}

.ModalDialog .Dialog .Header .X:active {
  background: #fd3d0d;
}

.ModalDialog .Dialog .Content {
  padding: 1rem;
}

.ModalDialog .Dialog .Footer {
  display: flex;
  justify-content: flex-end;
  padding: 0.25rem;
}

.ModalDialog .Dialog .Footer button {
  margin: 0.25rem;
  padding: 0.25rem;
}
