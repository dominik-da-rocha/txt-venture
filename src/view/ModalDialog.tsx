import "./ModalDialog.css";
import React from "react";
import Component, { ComponentProps, ComponentState } from "./Component";

export interface ModalDialogProps extends ComponentProps {
  title: string;
  onOkay?: () => void;
  onCancel?: () => void;
}

export interface ModalDialogState extends ComponentState {}

export class ModalDialog extends Component<ModalDialogProps, ModalDialogState> {
  constructor(props: ModalDialogProps) {
    super(props, {});
  }

  render() {
    return (
      <div id={this.props.id} className={this.className("ModalDialog")}>
        <div className="Dialog">
          <div className="Header">
            <span>{this.props.title}</span>
            <span
              className="X"
              id={this.makeId("x")}
              onClick={this.props.onCancel}
            >
              X
            </span>
          </div>
          <div className="Content">{this.props.children}</div>
          <div className="Footer">
            {this.props.onOkay ? (
              <button id={this.makeId("okay")} onClick={this.props.onOkay}>
                Okay
              </button>
            ) : undefined}
            {this.props.onCancel ? (
              <button id={this.makeId("cancel")} onClick={this.props.onCancel}>
                Cancel
              </button>
            ) : undefined}
          </div>
        </div>
      </div>
    );
  }
}

export default ModalDialog;
