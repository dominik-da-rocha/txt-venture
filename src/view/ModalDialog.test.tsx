import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Renderer from "react-test-renderer";
import React from "react";
import ModalDialog from "./ModalDialog";

Enzyme.configure({ adapter: new Adapter() });

describe("ModalDialog", () => {
  it("renders without crashing", () => {
    expect(() => {
      shallow(
        <ModalDialog
          id="test"
          title="Title"
          onOkay={() => {}}
          onCancel={() => {}}
        />
      );
    }).not.toThrow();
  });

  it("renders snapshot", () => {
    const tree = Renderer.create(
      <ModalDialog
        id="test"
        title="Title"
        onOkay={() => {}}
        onCancel={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders no okay callback", () => {
    const tree = Renderer.create(
      <ModalDialog
        id="test"
        title="Title"
        onCancel={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders no cancel callback", () => {
    const tree = Renderer.create(
      <ModalDialog
        id="test"
        title="Title"
        onOkay={() => {}}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });  
});
