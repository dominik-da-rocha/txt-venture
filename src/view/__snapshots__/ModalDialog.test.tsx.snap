// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDialog renders no cancel callback 1`] = `
<div
  className="Component ModalDialog"
  id="test"
>
  <div
    className="Dialog"
  >
    <div
      className="Header"
    >
      <span>
        Title
      </span>
      <span
        className="X"
        id="test-x"
      >
        X
      </span>
    </div>
    <div
      className="Content"
    />
    <div
      className="Footer"
    >
      <button
        id="test-okay"
        onClick={[Function]}
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalDialog renders no okay callback 1`] = `
<div
  className="Component ModalDialog"
  id="test"
>
  <div
    className="Dialog"
  >
    <div
      className="Header"
    >
      <span>
        Title
      </span>
      <span
        className="X"
        id="test-x"
        onClick={[Function]}
      >
        X
      </span>
    </div>
    <div
      className="Content"
    />
    <div
      className="Footer"
    >
      <button
        id="test-cancel"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;

exports[`ModalDialog renders snapshot 1`] = `
<div
  className="Component ModalDialog"
  id="test"
>
  <div
    className="Dialog"
  >
    <div
      className="Header"
    >
      <span>
        Title
      </span>
      <span
        className="X"
        id="test-x"
        onClick={[Function]}
      >
        X
      </span>
    </div>
    <div
      className="Content"
    />
    <div
      className="Footer"
    >
      <button
        id="test-okay"
        onClick={[Function]}
      >
        Okay
      </button>
      <button
        id="test-cancel"
        onClick={[Function]}
      >
        Cancel
      </button>
    </div>
  </div>
</div>
`;
